src/Dialog.elm

```diff
@@ -105,6 +106,14 @@ view maybeConfig =
             ]
 
 
+overlayCloser : Maybe (Config msg) -> List (Html.Attribute msg)
+overlayCloser maybeConfig =
+    maybeConfig
+        |> Maybe.andThen .closeMessage
+        |> Maybe.andThen (\closeMessage -> Just [ onClick closeMessage ])
```

Comment:
This could be replaced with
```
|> Maybe.map (\closeMessage -> [ onClick closeMessage ]) 
```
or even
```
|> Maybe.map (List.singleton << onClick) 
```
if you want to be fancy :slightly_smiling_face:

Reply:
Took your fancy approach. ;)